engine/src/main/java/org/destinationsol/common/SolNullOptionalException.java

```diff
@@ -5,27 +5,19 @@
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- * http://www.apache.org/licenses/LICENSE-2.0
+ *      http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package org.destinationsol.common;
 
-import java.lang.annotation.Documented;
-import java.lang.annotation.ElementType;
-import java.lang.annotation.Target;
-
-/**
- * Used to indicate that the value of a parameter/variable is not allowed to be null.
- * When used on a method, it means that the method should never return a null value.
- */
-@Documented
-@Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.LOCAL_VARIABLE, ElementType.METHOD})
-public @interface NotNull {
-
-}
\ No newline at end of file
+public class SolNullOptionalException extends SolDescriptiveException {
+    public SolNullOptionalException() {
+        super("Some object, which contained a null where by design it shouldn't have, caused a crash to the game. Please report this; there is some major design flaw within the game."
```

Comment:
`caused a crash to the game` -> `caused the game to crash`
`Please report this; there is some major design flaw within the game` -> Rephrase this more.. genially.